modules/snooker/src/test/ts/browser/TableSizeTest.ts

```diff
@@ -0,0 +1,111 @@
+  const table = Element.fromHtml<HTMLTableElement>(percentTableHtml);
+  Insert.append(container, table);
+  Insert.append(Body.body(), container);
+
+  const sizing = TableSize.getTableSize(table);
+  const warehouse = Warehouse.fromTable(table);
+
+  Assert.eq('Width should be 75', 80, sizing.width());
+  Assert.eq('Pixel width should be 600px', 400, sizing.pixelWidth());
+  Assert.eq('Cell widths should be 50% each', [ 50, 50 ], sizing.getWidths(warehouse, ResizeDirection.ltr, sizing));
+  Assert.eq('Cell min width should be at least 10px in percentage (2.5%)', true, sizing.minCellWidth() >= 2.5);
+
+  fc.assert(fc.property(fc.integer(-390, 390), fc.nat(100), (delta, colWidth) => {
+    const deltaPercent = delta / 400 * 100;
+    Assert.eq('Cell delta should be the same, but in percentage', deltaPercent, sizing.getCellDelta(delta));
+    Assert.eq('Single column delta width should be 100% - percentage width', [ 100 - colWidth ], sizing.singleColumnWidth(colWidth, delta));
+  }));
+
+  sizing.setTableWidth(table, [ 50, 50 ], -25);
+  Assert.eq('Table width after resizing is 25% less of the original 80%', '60%', Css.getRaw(table, 'width').getOr(''));
+
+  const cell = SelectorFind.descendant<HTMLTableCellElement>(table, 'td').getOrDie();
+  sizing.setElementWidth(cell, 25);
+  Assert.eq('Cell width after resizing is 25%', '25%', Css.getRaw(cell, 'width').getOr(''));
+
+  Remove.remove(container);
+});
+
+UnitTest.test('TableSize.noneSizing', () => {
+  const table = Element.fromHtml<HTMLTableElement>(noneTableHtml);
+  Insert.append(Body.body(), table);
+
+  const sizing = TableSize.getTableSize(table);
+  const warehouse = Warehouse.fromTable(table);
+  const width = Width.get(table);
+  const cellWidth = SelectorFind.descendant<HTMLTableCellElement>(table, 'td').map(Width.get).getOrDie();
+
+  Assert.eq('Width should be the computed size of the table', width, sizing.width());
+  Assert.eq('Pixel width should be the computed size of the table', width, sizing.pixelWidth());
+  Assert.eq('Cell widths should be the computed size of the cell', [ cellWidth, cellWidth ], sizing.getWidths(warehouse, ResizeDirection.ltr, sizing));
+  Assert.eq('Cell min width should be at least 10px', true, sizing.minCellWidth() >= 10);
+
+  fc.assert(fc.property(fc.integer(-390, 390), fc.integer(400, 1000), (delta, colWidth) => {
+    Assert.eq('Cell delta should be identity', delta, sizing.getCellDelta(delta));
+    Assert.eq('Single column delta width should be the delta', [ delta ], sizing.singleColumnWidth(colWidth, delta));
+  }));
+
+  sizing.setTableWidth(table, [ cellWidth - 10, cellWidth - 10 ], -20);
+  Assert.eq('Table width after resizing is unchanged', '', Css.getRaw(table, 'width').getOr(''));
+
+  const cell = SelectorFind.descendant<HTMLTableCellElement>(table, 'td').getOrDie();
+  sizing.setElementWidth(cell, 20);
+  Assert.eq('Cell width after resizing is unchanged', '', Css.getRaw(cell, 'width').getOr(''));
```

Comment:
Are you expecting `Css.getRaw(...)` to return `Option.none()` here?

If so

```
import { OptionInstances } from '@ephox/katamari';
const tOption = OptionInstances.tOption;

Assert.eq('...', Option.none(), Css.getRaw(cell, 'width'), tOption());
```

Reply:
Oh nice, I didn't know about that, thanks!

Reply:
Also not really aiming to test `Option`'s as the output, just the raw CSS value instead which is represented as an `Option`. This will still be better than the current way I'm doing that though.

Reply:
The only difference really is if `Css.getRaw(cell, 'width')` can possibly return `some('')`, is your test still valid?

Reply:
No, that should be a failure if it does that.

Reply:
Changed to use `tOption` to compare.